store/src/java/com/zimbra/cs/service/account/ModifyPrefs.java

```diff
@@ -76,14 +87,62 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
 
         HashMap<String, Object> prefs = new HashMap<String, Object>();
         Map<String, Set<String>> name2uniqueAttrValues = new HashMap<String, Set<String>>();
-        for (KeyValuePair kvp : request.listKeyValuePairs(AccountConstants.E_PREF, AccountConstants.A_NAME)) {
+        for (KeyValuePair kvp : request.listKeyValuePairs(AccountConstants.E_PREF,
+            AccountConstants.A_NAME)) {
             String name = kvp.getKey(), value = kvp.getValue();
             char ch = name.length() > 0 ? name.charAt(0) : 0;
             int offset = ch == '+' || ch == '-' ? 1 : 0;
             if (!name.startsWith(PREF_PREFIX, offset))
-                throw ServiceException.INVALID_REQUEST("pref name must start with " + PREF_PREFIX, null);
+                throw ServiceException.INVALID_REQUEST("pref name must start with " + PREF_PREFIX,
+                    null);
 
-            AttributeInfo attrInfo = AttributeManager.getInstance().getAttributeInfo(name.substring(offset));
+            if (Provisioning.A_zimbraPrefAccountProfileImage.equals(name)) {
+                String attachId = value;
+                Upload up = FileUploadServlet.fetchUpload(zsc.getAuthtokenAccountId(), attachId,
```

Comment:
Can we extract this to a utility method.

Reply:
Done